name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TastytradePDFExtractor.java

```diff
@@ -0,0 +1,529 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import name.abuchen.portfolio.datatransfer.ExtractorUtils;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Block;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.DocumentType;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.ParsedData;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Transaction;
+import name.abuchen.portfolio.model.AccountTransaction;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.util.OccOsiSymbology;
+
+@SuppressWarnings("nls")
+public class TastytradePDFExtractor extends AbstractPDFExtractor
+{
+    private List<BuySellEntry> m_optionTradesInImport = new ArrayList<>();
+
+    public TastytradePDFExtractor(Client client)
+    {
+        super(client);
+        addBankIdentifier("TASTYWORKS");
+        addBankIdentifier("tastytrade");
+        addBankIdentifier("Apex Clearing Corporation");
+
+        addSummaryStatementBuySellTransaction(); // Purchase and sale ( multiple settlements )
+        addDepotStatementTransaction(); // Securities account transactions ( Settlement account )
+        addDividendTransaction(); // Dividends
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "tastytrade, Inc.";
+    }
+
+    private void addDividendTransaction()
+    {
+        DocumentType type = new DocumentType("A[\\s]+C[\\s]+C[\\s]+O[\\s]+U[\\s]+N[\\s]+T[\\s]+"
+                        + "S[\\s]+T[\\s]+A[\\s]+T[\\s]+E[\\s]+M[\\s]+E[\\s]+N[\\s]+T");
+        this.addDocumentTyp(type);
+
+        Block firstRelevantLine = new Block("^DIVIDEND[\\s]+[\\d]{2}/[\\d]{2}/[\\d]{2}[\\s]+[A-Z][\\s]+.*$");
+        type.addBlock(firstRelevantLine);
+
+        Transaction<AccountTransaction> pdfTransaction = new Transaction<>();
+        pdfTransaction.subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DIVIDENDS);
+            entry.setCurrencyCode("USD");
+            return entry;
+        });
+        firstRelevantLine.set(pdfTransaction);
+
+        pdfTransaction
+            // DIVIDEND 06/28/23 M PROSHARES ULTRAPRO S&P500
+            // 0.145499 29.10
+            .section("date", "name", "price", "amount")
+            .match("DIVIDEND[\\s]+(?<date>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+[A-Z][\\s]+(?<name>[\\/\\&\\s\\w]+)[\\s]+"
+                            + "(\\$)?(?<price>[\\.,\\d]+)[\\s]+(\\$)?(?<amount>[\\.,\\d]+)")
+            .assign((t, v) -> {
+                v.put("currency", "USD");
+                t.setDateTime(asDate(v.get("date")));
+                t.setAmount(asAmount(v.get("amount"), "en", "US"));
```

Comment:
Remove all this "en", "US"
```suggestion
                t.setAmount(asAmount(v.get("amount")));
```

and add to the end of the file
```
    @Override
    protected long asAmount(String value)
    {
        return ExtractorUtils.convertToNumberLong(value, Values.Amount, "en", "US");
    }

    @Override
    protected long asShares(String value)
    {
        return ExtractorUtils.convertToNumberLong(value, Values.Share, "en", "US");
    }

    @Override
    protected BigDecimal asExchangeRate(String value)
    {
        return ExtractorUtils.convertToNumberBigDecimal(value, Values.Share, "en", "US");
    }
```

look for example [here](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/CreditSuisseAGPDFExtractor.java).

Reply:
- Added `protected long asAmount(String value)` and `protected long asShares(String value)`
- `protected BigDecimal asExchangeRate(String value)` is not used.

Done ✔